### ISSUE TITLE
False positive `reimported` related to import alias to avoid name clash

### ISSUE THREAD

**Issue by e-gebes** [pylint-dev/pylint#11353]
### Bug description

```python
# Simple package layout:

#   pkg/
#       __init__.py
#       utils/
#           __init__.py
#           foo.py
#           json.py

# Content of json.py:

ONE = 1

# Content of foo.py:

import json
import re
import re as re1                  # triggers `reimported` -> OK

from . import json as my_json     # triggers `reimported` -> false positive

from .json import ONE             # does not trigger -> OK
```

### Configuration

```ini

```

### Command used

```shell
pylint pkg --disable=all --enable=reimported
```

### Pylint output

```python
************* Module pkg.utils.foo
pkg/utils/foo.py:3:0: W0404: Reimport 're' (imported line 2) (reimported)
pkg/utils/foo.py:5:0: W0404: Reimport 'json' (imported line 1) (reimported)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

I would expect a warning for line 3 (as reported), but not for line 5, as it is just the same name for a module but it is a different module.

The own module "json" could just be renamed to avoid any name clashes with the std lib module. Still, the `reimported` checker should not mistake different modules of the same name as _being_ the same module. After all, using an import alias is a common way to resolve name clashes.

### Pylint version

```shell
pylint 4.0.7
astroid 4.0.4
Python 3.14.7 (main, Aug  5 2026, 15:45:41) [Clang 22.1.3 ]
```

### OS / Environment

uv virtual env

### Additional dependencies

```python

```
